torch/lib/c10d/Utils.hpp

```diff
@@ -304,6 +304,8 @@ while (true) {                                                \
   if (!(success_cond)) {                                      \
     if (errno == EINTR) {                                     \
       continue;                                               \
+    } else if (errno == EAGAIN || errno == EWOULDBLOCK) {     \
```

Comment:
This is trying to improve the error message. It was showing: `RuntimeError: resource temporarily unavailable`

Reply:
Sounds good. This is still a stop gap though -- we should add a proper exception hierarchy here, with links to the online docs or knowledge base to let people figure out what's going on.